src/server.rs

```diff
@@ -181,27 +176,23 @@ async fn delete(
 
 /// Expose version endpoint
 #[get("/version")]
-async fn version(
-    request: HttpRequest,
-    config: web::Data<RwLock<Config>>,
-) -> Result<HttpResponse, Error> {
+#[actix_web_grants::protect("TokenType::Auth", ty = TokenType, error = unauthorized_error)]
+async fn version(config: web::Data<RwLock<Config>>) -> Result<HttpResponse, Error> {
     let config = config
         .read()
         .map_err(|_| error::ErrorInternalServerError("cannot acquire config"))?;
-    let connection = request.connection_info().clone();
-    let host = connection.realip_remote_addr().unwrap_or("unknown host");
-    let tokens = config.get_tokens(TokenType::Auth);
-    auth::check(host, request.headers(), tokens)?;
     if !config.server.expose_version.unwrap_or(false) {
         warn!("server is not configured to expose version endpoint");
-        Err(error::ErrorForbidden("endpoint is not exposed\n"))?;
+        Err(error::ErrorNotFound("endpoint is not exposed\n"))?;
```

Comment:
If we don't want to expose anything, should we remove the message as well and just return 404?

Reply:
It makes a lot of sense to me